### PR TITLE
Prevent assigning PR authors to their own PRs

### PR DESCRIPTION
**Pull request by marcos-arranz**
This PR fixes codeassigner assigning the author as a reviewer, and also a possible infinite loop.

### CODE REVIEW

File: src/assign.ts [hunk 1]
```diff
@@ -187,6 +187,11 @@ export const selectReviewers = async (
       }
       debug(`Found random team member: ${randomTeamMember}.`)
 
+      if (randomTeamMember && randomTeamMember === author) {
+        debug(`'${randomTeamMember}' is the author '${randomTeamMember}'. Skipping.`)
+        continue
+      }
```
Comment (andersfischernielsen) [inline src/assign.ts:193]: We already assert that `randomTeamMember` isn't `undefined | null` in L:181 😄 

```suggestion
      if (randomTeamMember === author) {
        debug(`'${randomTeamMember}' is the author '${author}'. Skipping.`)
        continue
      }
```

File: src/assign.ts [hunk 2]
```diff
@@ -153,7 +152,8 @@ export const selectReviewers = async (
     const randomFileOwner = randomize(stack.find(owner => owner.pattern === randomFile)?.owners)?.shift()
     debug(`Selected random file owner: ${randomFileOwner}`)
     const randomGlobalCodeowners = randomize(globalCodeowners)
-    const selected = randomFileOwner ?? randomGlobalCodeowner(randomGlobalCodeowners)
+    const selected = randomFileOwner ?? randomGlobalCodeowners?.[0]
+    if (selected && !isTeam(selected)) randomGlobalCodeowners?.shift()
```
Comment (andersfischernielsen) [inline src/assign.ts:156]: `selected` being `undefined | null` is already handled in the following two checks and we already `break` if `!selected`, so this check is redundant 😄 

```suggestion
    if (!isTeam(selected)) randomGlobalCodeowners?.shift()
```

Comment (andersfischernielsen) [reply src/assign.ts:156]: I'd rather we change `isTeam` to handle a `string?` parameter 😄 to keep the logic consistent

File: src/assign.test.ts [hunk 3]
```diff
@@ -524,6 +524,51 @@ describe('Reviewer selection', () => {
...
+    }
+
+    const result = await selectReviewers(filesChanged, codeowners, {}, options)
+
+    expect(result).not.toBeNull()
+    expect(result.count).toEqual(2)
+    expect(result.users.every(name => name !== options.author)).toBeTruthy()
+  })
+
+  it('does not select an author from CODEOWNERS when owner is a team', async () => {
```
Comment (andersfischernielsen) [inline src/assign.test.ts:546]:
```suggestion
  it('does not select individual authors from CODEOWNERS when owners are teams', async () => {
```

File: src/assign.test.ts [hunk 4]
```diff
@@ -524,6 +524,51 @@ describe('Reviewer selection', () => {
     expect(result.users.every(name => name !== options.author)).toBeTruthy()
   })
 
+  it('does not select an author from CODEOWNERS with assignIndividuals true', async () => {
```
Comment (andersfischernielsen) [inline src/assign.test.ts:527]: Doesn't this break the behavior of the Action? 

If I have a repository where the `CODEOWNERS` looks like this:

```
* @some-org/cool-team
```

where the team consists of the members `@some-username` and I set `assignIndividuals: true`

then I'd expect the result of the Action running to be assigning `@some-username` to the PR and NOT the global team name. It doesn't look like that's the case anymore 🤔

**Comment by andersfischernielsen** [comment]
I'm a bit scared we're 
- breaking the expected functionality of the Action
- re-introducing logic that's been problematic in the past wrt. the infinite looping - I spent a decent chunk of time avoiding that exact loop a while back

with this 😄

File: src/assign.test.ts (hunk 4 above)
Comment (marcos-arranz) [reply src/assign.test.ts]: No, that's should not be the case. The behaviour to select an individual or not hasn't been changes. I have added a test case with that example to cover it.

**Comment by marcos-arranz** [comment]
> I'm a bit scared we're
* breaking the expected functionality of the Action
* re-introducing logic that's been problematic in the past wrt. the infinite looping - I spent a decent chunk of time avoiding that exact loop a while back
with this 😄

Thanks for your comments @andersfischernielsen, I have addressed most of them. Regarding your fears, I have added a test case to cover the first point. Regarding the second case, could you point me which code can fall into a infinite loop? I have added some code to prevent, because there were some conditions were it surface it as in this test run https://github.com/pleo-io/assign-random-codeowners/actions/runs/4146429489/jobs/7172152599

**Review by andersfischernielsen — APPROVED** [review]
LGTM! 🍰

**Pull request merged**